### PR TITLE
T-13513 Add grace period

### PR DESCRIPTION
**Pull request by pablocampogo**
- Adds grace period feature for not showing apps as rate limited until certain hours passes
- Apps that have never passed their limits are sent to the PHD endpoint to set the `FirstDateSurpassed` field

Dependency: https://github.com/pokt-foundation/pocket-http-db/pull/35

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -156,9 +193,20 @@ func (c *Cache) SetCache() error {
 			continue
 		}
 
-		if relayCount.Count.Failure+relayCount.Count.Success >= int64(appLimit.DailyLimit) {
+		count := int(relayCount.Count.Failure + relayCount.Count.Success)
```
Comment (adshmh) [inline cache/cache.go:196]: We should only count successful relays.

File: cache/cache_test.go [hunk 2]
```diff
@@ -58,4 +61,16 @@ func TestCache_SetCacheFailure(t *testing.T) {
 
 	err = cache.SetCache()
 	c.Equal(errUnexpectedStatusCodeInRelays, err)
+
+	mock.AddMockedResponseFromFile(http.MethodGet, fmt.Sprintf("%s%s", httpDBURL, appLimitsEndpoint),
+		http.StatusOK, "../samples/apps_limits.json")
```
Comment (adshmh) [inline cache/cache_test.go:66]: Not a blocker, but we should use a `test_data` directory to store all test data files.

File: samples/apps_relays.json [hunk 3]
```diff
@@ -25,5 +25,14 @@
     "From": "2022-08-12T00:00:00Z",
     "To": "2022-08-13T00:00:00Z",
     "Application": "4e28a9ccc2997538dadae2a9fe03de52e9b309b355e75b1f368062frsf4nsctv"
+  },
+  {
+    "Count": {
+      "Success": 254673,
```
Comment (adshmh) [inline samples/apps_relays.json:31]: We need to update the tests to make sure failed relays are not counted towards the limit

**Review by adshmh — CHANGES_REQUESTED** [review]

File: cache/cache_test.go (hunk 2 above)
Comment (pablocampogo) [reply cache/cache_test.go:66]: I'll do that on other PR, it changes many things

**Review by adshmh — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**